users/views.py

```
@@ -67,6 +68,26 @@ def validate_phone(request):
     return PhoneDevice.send_otp_httpresponse(phone_number=user.phone_number, user=user)
 
 
+@api_view(["POST"])
+def validate_firebase_id_token(request):
+    user = request.user
+    id_token = request.data.get("id_token")
+    if not id_token:
+        return JsonResponse({"error": ErrorCodes.MISSING_TOKEN}, status=400)
+    try:
+        decoded_token = auth.verify_id_token(id_token)
+    except Exception:
+        return JsonResponse({"error": ErrorCodes.FAILED_VALIDATING_TOKEN}, status=400)
+
+    if not decoded_token.get("uid"):
```

Why do we need to check this?

---

This was added as a sanity check to make sure that nothing went wrong in the request and that Firebase was able to successfully validate the token with a user. This is not strictly necessary, as I believe `verify_id_token` should throw an exception in most cases if something went wrong, but this was just for that _extra_ peace of mind. 

I don't have strong feelings over this check, so happy to remove this if you think it's just bloating the function unecessarily.